From 65a37d591ce39d630c6e94b222e02439fd171351 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Wed, 11 Dec 2024 23:29:38 +0800
Subject: [PATCH 277/300] AOSCOS: MIPS: ls7a_fan: use register addresses in
 loongson.h

Original driver code expects register definition in loongson-pch.h (where
LS7A register definitions were never merged upstream), which are now
revised and stored in loongson.h.

Miscellaneous register base (LS7A_MISC_REG_BASE) starts at:

  LOONGSON_PCILO0_BASE + 0x80000

According to legacy loongson-pch.h from Lemote. Revise definition for
LS7A_PWM_REG_BASE to be:

  LOONGSON_PCILO0_BASE (LS7A_PCH_REG_BASE in loongson-pch.h)
    + 0x80000 = LS7A_MISC_REG_BASE
  LS7A_MISC_REG_BASE + 0x2000 = LS7A_PWM_REG_BASE

Fixes: "AOSCOS: MIPS: ls7a_fan: use register addresses in loongson.h"
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 drivers/platform/mips/ls7a_fan.c | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/drivers/platform/mips/ls7a_fan.c b/drivers/platform/mips/ls7a_fan.c
index cc188d4d0576..e9cf2dded39e 100644
--- a/drivers/platform/mips/ls7a_fan.c
+++ b/drivers/platform/mips/ls7a_fan.c
@@ -9,10 +9,16 @@
 #include <asm/io.h>
 
 #include <boot_param.h>
-#include <loongson-pch.h>
+#include <loongson.h>
 #include <loongson_hwmon.h>
 
-#define LS7A_PWM_REG_BASE              (void *)TO_UNCAC(LS7A_MISC_REG_BASE + 0x20000)
+/**
+ * Miscellaneous register base (LS7A_MISC_REG_BASE) starts at
+ * LOONGSON_PCILO0_BASE + 0x80000, according to legacy loongson-pch.h
+ * from Lemote.
+ */
+#define LS7A_PWM_REG_BASE              (void *)TO_UNCAC(LOONGSON_PCILO0_BASE + \
+							0x80000 + 0x20000)
 
 #define LS7A_PWM0_LOW                   0x004
 #define LS7A_PWM0_FULL                  0x008
-- 
2.48.1

